File: Leetcode/JAVA.Again/_292_nimGame.py



class Solution:
    store = {0 : True , 1: True, 2: True}
    def canWinNim(self, n: int) -> bool:
        global store 

        if n <= 3 or store[n-1] == False or store[n-2] == False or store[n-3] == False:
            store[n] = True
            return True

        else :
            ans = not self.canWinNim(n-3) and not self.canWinNim(n-2) and not self.canWinNim(n-1)
            store[n] = ans
            return ans

File: Leetcode/2023/292NimGame.java
class Solution {

    public boolean canWinNim(int n) {
        boolean canWin = true ;
        for(int i = 3; i <=n; )  {
            if(canWin) i+= 1;
            else i+=2 ;
            canWin = !canWin ;
        }
        return canWin ;

    }

//n = 1348820612    STACK OVER
}